doc/mgmt/Show_Running_Configuration_HLD.md

```diff
@@ -0,0 +1,495 @@
+
+NA
+
+
+
+#### 1 Feature Overview
+
+This feature provides an infrastructure for apps to add
+their current system configuration to the output of the "show
+running-configuration" command. This command is available at the exec
+level on the SONiC management framework CLI.
+
+#### 1.1 Functional requirements
+
+The 'show running-configuration' command will display the  configuration done
+through the CLI interface and which is stored on the system.
+
+For this command to show the entire running configuration of the system,
+all apps should support configuration through CLI interface.
+
+Any configuration done using GNMI or REST or other management
+interface which is stored and has a corresponding CLI
+command will be displayed as part of this new command.
+
+The command output can be copied and pasted to be used as
+configuration on another SONiC CLI session of the same or another
+system.
+
+The order of commands displayed in the "show running-configuration" will be
+in accordance to the order of dependency between CLI views.
+
+If configuration default values are stored, they will be shown as
+part of "show running-configuration".
+
+The infrastructure will provide functionality to filter and display
+the configuration based on the CLI views.
+
+A plugin in form of callback will be provided for apps to do their own database retrieval and
+rendering. Apps will return the list of commands to the infra which will then be
+added to the final list of commands to be displayed.
+
+#### 1.2 Restrictions
+
+Apps should not use this infrastructure to add configuration to 'show running-configuration' command for which there does not exist a CLI command.
+
+System state information should not be added to this command.
+
+The ordering between commands in the same view or between views is not supported. Hence some set of configuration will not work for copy-paste operation.
+
+#### 1.3 Command syntax
+
+###### 1.3.1 show running-configuration
+
+An exec level command to display the current configuration of the
+system.
+
+Example:
+
+sonic\#\>show running-config  
+interface Vlan 5  
+\!  
+interface PortChannel 5 min-links 3 mode active  
+\!  
+interface Ethernet0  
+&nbsp; no shutdown  
+&nbsp; mtu 5000  
+&nbsp; nat-zone 2  
+&nbsp; ip address 10.11.203.35/8  
+\!  
+interface Ethernet4  
+&nbsp; no shutdown  
+&nbsp; mtu 5000  
+&nbsp; switchport access vlan 5  
+&nbsp; switchport trunk allowed vlan 6  
+\!  
+router bgp 11 vrf VrfBlack  
+&nbsp; listen range 8.8.8.0/24 peer-group PG1  
+\!  
+router bgp 5 vrf VrfBlue  
+&nbsp; router-id 6.6.6.6  
+&nbsp; graceful-restart enable  
+&nbsp; graceful-restart preserve-fw-state  
+&nbsp; graceful-restart restart-time 500  
+&nbsp; graceful-restart stalepath-time 600  
+
+###### 1.3.2 show configuration
+
+This command displays the configuration pertaining to the view at
+which it is executed.
+
+Example:
+
+sonic(conf)\# interface PortChannel 5  
+sonic(conf-if-po5)\#show configuration  
```

Comment:
This is not a Cisco thing, right? Cisco instead seems to have "show running-config <feature_name>" commands. Wouldn't this be more industry-standard?